### PR TITLE
Add rake task to generate access tokens for all users

### PR DESCRIPTION
**Pull request by dmitrytrager**
According to: https://redmine.smartdata.solutions/issues/6162

### CODE REVIEW

**Comment by dmitrytrager** [comment]
@t-harps every created AccessToken will have "token" field generated and we need to keep these tokens on app side in order to pass it within request for data.

How do you think we should collect these tokens?

**Comment by t-harps** [comment]
@dmitrytrager Can I not find the AccessToken using the user id + application id?

**Comment by dmitrytrager** [comment]
Ok, will think about possibilities here.

**Comment by dmitrytrager** [comment]
Even if I search for AccessToken using scopes and expiration time within my PR, I think you can find it just by user and application. But I'm not sure it's the best option to expose endpoint to the whole world even if it would be protected (it should be protected with authentication filters, of course).

On the other hand, if we authenticate developer on api request (and we really do it), we can send access token for corresponding application for every developer, not only for default application.

What do you think @t-harps ?

**Comment by t-harps** [comment]
@dmitrytrager I'm not sure I understand. 

Are you saying that when a developer makes a request, we can attach the access token to the request before it hits the doorkeeper auth stage? I don't think this is what you're saying but might be an option?

**Comment by dmitrytrager** [comment]
I was thinking that we could return the access token from the get user endpoint for all apps. But now I see there is only copo app allowed to bypass doorkeeper authorization.

Nevermind. Will inlude token in non-public info for users.

**Comment by dmitrytrager** [comment]
@t-harps please, review

File: lib/tasks/oauth.rake [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    puts "Adding access tokens for all users:"
+
+    User.find_each(batch_size: 100) do |user|
+      Doorkeeper::AccessToken.find_or_create_for(
+        default_application,
+        user.id,
+        "public",
+        7200,  # default value from server config settings
+        true)  # not a default value, enabling refresh tokens here
+
```
Comment (t-harps) [inline lib/tasks/oauth.rake]: Need to complete the approval here also I think, i.e.
```
developer = Developer.default(mobile: true)
developer.approvals.find_by(user_id: current_resource_owner.id).tap do |approval|
  approval ||= Approval.add_developer(current_resource_owner, developer)
  approval.update(status: "complete")
end
```
Or something along those lines.

File: app/models/user.rb [hunk 2]
```diff
@@ -143,18 +147,29 @@ def display_name
...
 
+  def copo_app_access_token
+    copo_app = Developer.default(mobile: true).oauth_application
+    return nil unless copo_app
+
+    Doorkeeper::AccessToken
+      .where(resource_owner_id: id)
+      .where(application_id: copo_app.id)
+      .first
+      &.token
```
Comment (t-harps) [inline app/models/user.rb]: Added &.token just in case access_token doesnt exist yet.

Comment (dmitrytrager) [inline app/models/user.rb]: :+1:

**Pull request merged**